sklearn/_build_utils/pre_build_helpers.py

```diff
@@ -73,10 +73,14 @@ def compile_test_program(code, extra_preargs=[], extra_postargs=[]):
                                       extra_preargs=extra_preargs,
                                       extra_postargs=extra_postargs)
 
-            # Run test program
-            # will raise a CalledProcessError if return code was non-zero
-            output = subprocess.check_output('./test_program')
-            output = output.decode(sys.stdout.encoding or 'utf-8').splitlines()
+            if "PYTHON_CROSSENV" not in os.environ:
+                # Run test program if not cross compiling
+                # will raise a CalledProcessError if return code was non-zero
+                output = subprocess.check_output('./test_program')
+                output = output.decode(
+                    sys.stdout.encoding or 'utf-8').splitlines()
+            else:
+                output = []
```

Comment:
Why do we skip running the test program? Does `crossenv` compile `test_program`?

Reply:
Yes, we can compile the `test_program`. We can't run it because the program is compiled for the `host` (eg: Arm macOS), but we re running on `build` (eg: Intel macOS)

Reply:
Thank you, that makes sense. Let's leave a comment stating that the host may not be able to run  `test_program`.